src/button-dropdown/__tests__/button-dropdown-items.test.tsx

```diff
@@ -419,6 +420,14 @@ const items: ButtonDropdownProps.Items = [
         wrapper.openDropdown();
         expect(wrapper.findItemById('i1')!.findAllByClassName(IconWrapper.rootSelector)).toHaveLength(2);
       });
+
+      it('should render badge when defined', () => {
```

Comment:
Should we add two more tests for the trigger badge rendering 1) badge is rendered when there are items with badges. 2) badge is not rendered if variant is not `icon`

Reply:
Added (1) but not (2) since that isn't relevant anymore if this is internal-only